e2e_report_test.go

```diff
@@ -75,3 +77,116 @@ func TestE2EReport_Aliases(t *testing.T) {
 
 	assert.Contains(t, rep.reported, "foo.Object().ContainsKey()")
 }
+
+const (
+	intSize = 32 << (^uint(0) >> 63) // 32 or 64
+	maxInt  = 1<<(intSize-1) - 1
+)
+
+func TestE2EReport_LineWidth(t *testing.T) {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/test", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(`{"foo":123}`))
+	})
+
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	splitWhiteSpaceOrDot := func(r rune) bool {
+		return unicode.IsSpace(r) || r == '.'
+	}
+
+	tests := []struct {
+		name             string
+		formatter        *DefaultFormatter
+		wantMaxLineWidth int
+	}{
+		{
+			name: "zero value",
+			formatter: &DefaultFormatter{
+				LineWidth: 0,
+			},
+			wantMaxLineWidth: defaultLineWidth,
+		},
+		{
+			name: "small positive value",
+			formatter: &DefaultFormatter{
+				LineWidth: 15,
+			},
+			wantMaxLineWidth: 15,
+		},
+		{
+			name: "very large positive value",
+			formatter: &DefaultFormatter{
+				LineWidth: 1000,
+			},
+			wantMaxLineWidth: 1000,
+		},
+		{
+			name: "negative value (no limit)",
+			formatter: &DefaultFormatter{
+				LineWidth: -1,
+			},
+			wantMaxLineWidth: maxInt,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rep := &recordingReporter{}
+
+			e := WithConfig(Config{
+				TestName: "TestExample",
+				BaseURL:  server.URL,
+				AssertionHandler: &DefaultAssertionHandler{
+					Formatter: tt.formatter,
+					Reporter:  rep,
+				},
+			})
+
+			e.GET("/test").
+				Expect().
+				JSON().
+				Object().
+				NotContainsValue(123)
+
+			var skip bool
+			for _, s := range strings.Split(strings.Trim(rep.reported, "\n"), "\n") {
+				// only assert first errors block and assertPath block in defaultFailureTemplate
+				if strings.HasPrefix(s, "assertion:") {
+					skip = false
+				}
```

Comment:
Do we really need this parsing?

Can we instead make very simple assertions, like:

* in this test, there should be no lines longer than N
* in this test, there should be at least one line longer than N

This way we still check that line limit, well, limits line width, and at the same time we don't bother how and where exactly it is applied.

Reply:
good idea, thank you